server/sonar-server/src/main/java/org/sonar/server/platform/platformlevel/PlatformLevel.java

```diff
@@ -123,6 +129,24 @@ protected void add(Object... objects) {
     }
   }
 
+  /**
+   * Add a component to container only if the server node is marked as "startupLeader" (cluster disabled
+   * or first node of the cluster to be started).
+   *
+   * @throws IllegalStateException if called from PlatformLevel1, when cluster settings are not loaded
+   */
+  protected void addIfStartupLeader(Object... objects) {
+    Optional<Cluster> cluster = getOptional(Cluster.class);
+    checkState(cluster.isPresent(), "Cluster settings not loaded yet");
+    if (cluster.get().isStartupLeader()) {
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) Call "cluster.isPresent()" before accessing the value. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS3655)